core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -525,14 +523,43 @@ private void syncRemoteChangeEventToLocal(
     final BsonDocument docFilter = getDocumentIdFilter(docConfig.getDocumentId());
     final BsonDocument localDocument = localColl.find(docFilter).first();
 
-    final BsonValue lastKnownRemoteVersion;
+    final BsonDocument lastKnownRemoteVersion;
```

Comment:
if it helps, name this `currentRemoteVersion` which will become `lastKnownRemoteVersion` in the config